### PR TITLE
[SIG-1973] Saga exception handling

### PR DESCRIPTION
**Pull request by janjaap**
This PR attaches an exception handler to any saga that is injected into a container component. This allows us to tap into any error that is rethrown from a saga, like so:

```jsx
export function* sagaGeneratorFunc() {
  try {
    // attempt to fetch from a resource
  } catch (error) {
    // do whatever needs to be done when an error is caught

    // important bit
    throw(error);
  }
}
```

or so:

```jsx
export function* sagaGeneratorFunc() {
  throw(error);
}
```

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
 module.exports = {
   collectCoverageFrom: [
+    'src/*.js',
     'src/**/*.{js,jsx}',
+    '!src/*.test.js',
```
Comment (spaces-ga) [inline jest.config.js]: Is this not already covered by the line beneath?

File: src/configureStore.test.js [hunk 2]
```diff
@@ -29,6 +28,32 @@ describe('configureStore', () => {
...
+      }
+
+      expect(dispatchSpy).not.toHaveBeenCalled();
+
+      store.runSaga(someGenerator);
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          payload: {
+            message: expect.any(String),
```
Comment (spaces-ga) [inline src/configureStore.test.js]: We know the message and the string at this point right?

File: jest.config.js (hunk 1 above)
Comment (janjaap) [reply jest.config.js]: No, `configureStore.js`, for instance, didn't show up in the coverage report.

File: src/configureStore.test.js (hunk 2 above)
Comment (janjaap) [reply src/configureStore.test.js]: Yes, but for the test it doesn't really matter which message has been sent, just that a message has been sent.

**Pull request merged**